art/estimators/estimator.py

```diff
@@ -56,7 +56,7 @@ def __init__(
         clip_values: Optional["CLIP_VALUES_TYPE"],
         preprocessing_defences: Union["Preprocessor", List["Preprocessor"], None] = None,
         postprocessing_defences: Union["Postprocessor", List["Postprocessor"], None] = None,
-        preprocessing: "PREPROCESSING_TYPE" = (0, 1),
+        preprocessing: Union["PREPROCESSING_TYPE", "Preprocessor"] = (0, 1),
```

Comment:
If I understand correctly, the type of "Preprocessor" is applied for preprocessing_defences while the type "PREPROCESSING_TYPE" is applied for the preprocessing param for standardizing the input. So why now you add this type for preprocessing param, is there any design change?

Reply:
Yes, that's correct. We are now internally converting the preprocessing tuple to an instance of Preprocessor. One reason is to enable full framework support through the preprocessing steps. Because of the conversion `get_params` is returning the Preprocessor instance instead of the tuple and therefore the argument `preprocessing` and especially `set_params` accept now also the Preprocessor type for `preprocessing`. The tuple remains for backward compatibility.